jobs/import_process/worker.go

```diff
@@ -0,0 +1,196 @@
+)
+
+func init() {
+	app.RegisterJobsImportProcessInterface(func(a *app.App) tjobs.ImportProcessInterface {
+		return &ImportProcessInterfaceImpl{a}
+	})
+}
+
+type ImportProcessInterfaceImpl struct {
+	app *app.App
+}
+
+type ImportProcessWorker struct {
+	name        string
+	stopChan    chan struct{}
+	stoppedChan chan struct{}
+	jobsChan    chan model.Job
+	jobServer   *jobs.JobServer
+	app         *app.App
+}
+
+func (i *ImportProcessInterfaceImpl) MakeWorker() model.Worker {
+	return &ImportProcessWorker{
+		name:        "ImportProcess",
+		stopChan:    make(chan struct{}),
+		stoppedChan: make(chan struct{}),
+		jobsChan:    make(chan model.Job),
+		jobServer:   i.app.Srv().Jobs,
+		app:         i.app,
+	}
+}
+
+func (w *ImportProcessWorker) JobChannel() chan<- model.Job {
+	return w.jobsChan
+}
+
+func (w *ImportProcessWorker) Run() {
+	mlog.Debug("Worker started", mlog.String("worker", w.name))
+
+	defer func() {
+		mlog.Debug("Worker finished", mlog.String("worker", w.name))
+		close(w.stoppedChan)
+	}()
+
+	for {
+		select {
+		case <-w.stopChan:
+			mlog.Debug("Worker received stop signal", mlog.String("worker", w.name))
+			return
+		case job := <-w.jobsChan:
+			mlog.Debug("Worker received a new candidate job.", mlog.String("worker", w.name))
+			w.doJob(&job)
+		}
+	}
+}
+
+func (w *ImportProcessWorker) Stop() {
+	mlog.Debug("Worker stopping", mlog.String("worker", w.name))
+	close(w.stopChan)
+	<-w.stoppedChan
+}
+
+func (w *ImportProcessWorker) doJob(job *model.Job) {
+	if claimed, err := w.jobServer.ClaimJob(job); err != nil {
+		mlog.Warn("Worker experienced an error while trying to claim job",
+			mlog.String("worker", w.name),
+			mlog.String("job_id", job.Id),
+			mlog.String("error", err.Error()))
+		return
+	} else if !claimed {
+		return
+	}
+
+	importFileName, ok := job.Data["import_file"]
+	if !ok {
+		appError := model.NewAppError("ImportProcessWorker", "import_process.worker.do_job.missing_file", nil, "", http.StatusBadRequest)
+		w.setJobError(job, appError)
+		return
+	}
+
+	importFilePath := filepath.Join(*w.app.Config().ImportSettings.Directory, importFileName)
+	if ok, err := w.app.FileExists(importFilePath); err != nil {
+		w.setJobError(job, err)
+		return
+	} else if !ok {
+		appError := model.NewAppError("ImportProcessWorker", "import_process.worker.do_job.file_exists", nil, "", http.StatusBadRequest)
+		w.setJobError(job, appError)
+		return
+	}
+
+	importFileSize, appErr := w.app.FileSize(importFilePath)
+	if appErr != nil {
+		w.setJobError(job, appErr)
+		return
+	}
+
+	importFile, appErr := w.app.FileReader(importFilePath)
+	if appErr != nil {
+		w.setJobError(job, appErr)
+		return
+	}
+	defer importFile.Close()
+
+	// TODO: improve this process by eliminating the need to unzip the import
+	// file locally and instead do the whole bulk import process in memory by
+	// streaming the import file.
+
+	// create a temporary dir to extract the zipped import file.
```

Comment:
I'd like to open a ticket before merging this so we can keep track of this.

Reply:
Do we have a ticket for this created?

Reply:
Now we do ([MM-30187](https://mattermost.atlassian.net/browse/MM-30187)). I'll update the comment to include it.